optuna_dashboard/_app.py

```diff
@@ -411,6 +412,45 @@ def save_study_note(study_id: int) -> BottleViewReturn:
         response.status = 204  # No content
         return {}
 
+    @app.post("/api/studies/<trial_id:int>/tell")
+    @json_api_view
+    def tell_trial(trial_id: int) -> BottleViewReturn:
+        s = request.json.get("state", None)
+        vs = request.json.get("values", None)
+
+        try:
+            values = [float(v) for v in vs] if vs is not None else vs
+        except ValueError:
+            response.status = 400  # Bad request
+            return {"reason": "You need to pass float castable values"}
+
+        str_to_state = {
+            "Complete": TrialState.COMPLETE,
+            "Pruned": TrialState.PRUNED,
+            "Fail": TrialState.FAIL,
+        }
+        if s not in str_to_state:
+            response.status = 400  # Bad request
+            return {
+                "reason": f"You passed {s} as a state, but only 'Complete', 'Pruned' and 'Fail'"
+                "are acceptable states."
+            }
+        if s == "Complete" and values is None:
+            response.status = 400  # Bad request
+            return {
+                "reason": "When you passed 'Complete' as a state, you also need to specify values."
+            }
+        state = str_to_state[s]
+
+        try:
+            storage.set_trial_state_values(trial_id, state, values)
+        except Exception as e:
+            response.status = 400  # Bad request
+            return {"reason": e.args}
```

Comment:
500 should be returned here.

```suggestion
            response.status = 500
            return {"reason": f"Internal server error: {e}"}
```